modules/navier_stokes/src/executioners/LinearAssemblySegregatedSolve.C

```diff
@@ -571,31 +613,63 @@ LinearAssemblySegregatedSolve::solve()
   unsigned int simple_iteration_counter = 0;
 
   // Assign residuals to general residual vector
-  const unsigned int no_systems = _momentum_systems.size() + 1 + _has_energy_system +
-                                  _has_solid_energy_system + _active_scalar_systems.size() +
-                                  _turbulence_systems.size();
+  std::vector<std::pair<unsigned int, Real>> ns_residuals;
+  std::vector<Real> ns_abs_tols;
+
+  std::vector<unsigned int> momentum_indices;
+  if (_solve_momentum)
+    for ([[maybe_unused]] const auto system_i : index_range(_momentum_systems))
+    {
+      momentum_indices.push_back(static_cast<unsigned int>(ns_residuals.size()));
+      ns_residuals.push_back(std::make_pair(0, 1.0));
+      ns_abs_tols.push_back(_momentum_absolute_tolerance);
+    }
 
-  std::vector<std::pair<unsigned int, Real>> ns_residuals(no_systems, std::make_pair(0, 1.0));
-  std::vector<Real> ns_abs_tols(_momentum_systems.size(), _momentum_absolute_tolerance);
-  ns_abs_tols.push_back(_pressure_absolute_tolerance);
+  auto pressure_index = libMesh::invalid_uint;
```

Comment:
If the static casts and use of `unsigned int` is so that you can leverage this guy, I'd say we should define a `Moose::invalid_size_t` instead